sled-agent/src/sim/storage.rs

```diff
@@ -1172,10 +1175,15 @@ impl Zpool {
                 end_port,
             )),
         );
+
         let DatasetContents::Crucible(crucible) = self
             .datasets
             .get(&id)
-            .expect("Failed to get the dataset we just inserted");
+            .expect("Failed to get the dataset we just inserted")
+        else {
+            panic!("just inserted this variant!");
```

Comment:
obnoxious turbo nit: feels like a place for
```suggestion
            unreachable!("just inserted this variant!");
```